node_modules/oae-resource/lib/actions.js

```diff
@@ -0,0 +1,724 @@
+var AuthzModel = require('oae-authz/lib/model');
+var AuthzPermissions = require('oae-authz/lib/permissions');
+var AuthzUtil = require('oae-authz/lib/util');
+var EmitterAPI = require('oae-emitter');
+var Invitation = require('oae-authz/lib/invitations/model').Invitation;
+var OaeUtil = require('oae-util/lib/util');
+var PrincipalsDAO = require('oae-principals/lib/internal/dao');
+var PrincipalsUtil = require('oae-principals/lib/util');
+var TenantsAPI = require('oae-tenants');
+var Validator = require('oae-authz/lib/validator').Validator;
+
+var ResourceActivity = require('oae-resource/lib/activity');
+var ResourceConstants = require('oae-resource/lib/constants').ResourceConstants;
+
+var log = require('oae-logger').logger('oae-resource-actions');
+
+var ResourceActions = module.exports = new EmitterAPI.EventEmitter();
+
+/**
+ * Validate and create a resource using the given `createFn` function, while performing the role
+ * changes afterward.
+ *
+ *  TODO:   Passing in a `createFn` is not an ideal pattern. Would be better to have a central
+ *          registry for handling CRUD operations on resources, however it is outscoped ATM
+ *
+ * @param  {Context}            ctx                         Standard context object containing the current user and the current tenant
+ * @param  {Object}             roles                       The roles, keyed by a share target expression (e.g., principal id, email address, or `email:userId` expression), whose value is the role the target should have on the resource
+ * @param  {Function}           createFn                    The function to use to create the actual resource
+ * @param  {Object}             createFn.err                An error that occurred while creating the resource, if any
+ * @param  {Resource}           createFn.resource           The first argument should always be the resource that was created (i.e., the resource that will have the roles associated to it)
+ * @param  {...Object}          createFn.additionalArgs     Any additional arguments you need to return when creating the resource (e.g., a content revision)
+ * @param  {Function}           callback                    Standard callback function
+ * @param  {Object}             callback.err                An error that occurred, if any
+ * @param  {...Object}          callback.createFnResults    The result(s) of the create function (e.g., the created resource), one after another as separate arguments
+ * @param  {MemberChangeInfo}   callback.memberChangeInfo   The member change info object that describes the resource members
+ * @param  {EmailChangeInfo}    callback.emailChangeInfo    The email change info object that describes the resource invitations
+ */
+var create = module.exports.create = function(ctx, roles, createFn, callback) {
+    var validator = new Validator();
+    validator.check(null, {'code': 400, 'msg': 'Only authenticated users can create a new resurce'}).isLoggedInUser(ctx);
```

Comment:
Small typo (`resource`) here and in a couple of other spots. (Yes, these are now the only things I can complain about)